docs/customization.md

```diff
@@ -186,6 +186,27 @@ def get_resolver_provider(ctx, req, include_sdists, include_wheels, sdist_server
     ...
 ```
 
+The `GenericProvider` is a convenient base class, or can be instantiated
+directly if given a `version_source` callable that returns an iterator of
+version values as `str` or `Version` objects.
```

Comment:
Why allow both? Since we want to start enforcing the use of `Version` and this is a new API we are introducing, we can enforce it without causing any breaking changes?

Reply:
Ease of use from the caller's perspective, since most of those will be plugins. https://en.wikipedia.org/wiki/Robustness_principle

Reply:
ahh make sense